irohad/subscription/subscription_engine.hpp

```diff
@@ -62,10 +72,26 @@ namespace iroha::subscription {
         std::unordered_map<EventKeyType, SubscriptionContext>;
 
     mutable std::shared_mutex subscribers_map_cs_;
+
+    /// Associative container with a lists of subscribers by the event key
     KeyValueContainer subscribers_map_;
+
+    /// Thread handlers dispatcher
     DispatcherPtr dispatcher_;
 
    public:
+    /**
+     * Stores Subscriber object to retrieve later notifications
+     * @tparam kTid Thread ID in which subscribers callback will be executed
+     * @param set_id subscription set id is a group identifier in multiple
+     * subscriptions
+     * @param key notification event key that will be listened by this
+     * subscriber
+     * @param ptr subscriber weak pointer
+     * @return a position in an internal container with subscribers(!!! it must
+     * be keep safe in a case the other subscriber will be deleted from this
```

Comment:
```suggestion
     * be kept safe in case the other subscriber will be deleted from this
```
But what does "kept safe" in this context mean? Like a `shared_ptr` that should not be deleted because of reference counting?